server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/lisk/wallets/liskbinancewallet/LskWallet.java

```diff
@@ -0,0 +1,157 @@
+package com.generalbytes.batm.server.extensions.extra.lisk.wallets.liskbinancewallet;
+
+import com.generalbytes.batm.server.extensions.Currencies;
+import com.generalbytes.batm.server.extensions.IWallet;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import si.mazi.rescu.HttpStatusIOException;
+import si.mazi.rescu.RestProxyFactory;
+
+import org.apache.commons.codec.binary.Hex;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.UnsupportedEncodingException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+
+import java.util.Random; 
+
+public class LskWallet implements IWallet {
+    private static final Logger log = LoggerFactory.getLogger("batm.master.LskWallet");
+
+    private String nonce;
+    private String accessHash;
+
+    private String address;
+    private String binanceApiKey;
+    private String binanceApiSecret; 
+ 
+    private LskBinanceAPI apiBinance;
+
+    public LskWallet(String address, String binanceApiKey, String binanceApiSecret) {
+        this.nonce = generateRandomString(8);
+
+        this.address = address;
+        this.binanceApiKey = binanceApiKey;
+        this.binanceApiSecret = binanceApiSecret; 
+
+        apiBinance = RestProxyFactory.createProxy(LskBinanceAPI.class, "https://api.binance.com");
+    }
+
+    @Override
+    public String getPreferredCryptoCurrency() {
+        return Currencies.LSK;
+    }
+
+    @Override
+    public Set<String> getCryptoCurrencies() {
+        Set<String> result = new HashSet<String>();
+        result.add(Currencies.LSK);
+        return result;
+    }
+
+    @Override
+    public String getCryptoAddress(String cryptoCurrency) {
+        if (!getCryptoCurrencies().contains(cryptoCurrency)) {
+            log.error("Cryptocurrency " + cryptoCurrency + " not supported.");
+            return null;
+        }
+        if (address != null) {
+            return address;
+        }
+
+        return null;
+    }
+
+    @Override
+    public BigDecimal getCryptoBalance(String cryptoCurrency) {
+        if (!getCryptoCurrencies().contains(cryptoCurrency)) {
+            log.error("Cryptocurrency " + cryptoCurrency + " not supported.");
+            return null;
+        }
+        try {
+
+            String query = "";
+            String timeStamp = String.valueOf(new Date().getTime());
+            query = "recvWindow=" + 5000 + "&timestamp=" + timeStamp; 
+ 
+            String signing = sign(query, binanceApiSecret);
+
+            final Map<String, Object> accountInfo = apiBinance.getCryptoBalance(this.binanceApiKey, String.valueOf(5000), timeStamp, signing);
```

Comment:
The same thing as I already mentioned. Don't use untyped `Map`s for results. Specify proper types for the API. Otherwise, every decimal can lose precise value information until it gets to `BigDecimal`.